docs/Compile.rst

```diff
@@ -697,55 +704,89 @@ during a build.  If you get a failure, you may miss important errors or warnings
+  other platforms.
+
+The generated documentation will be stored in ``docs/html`` in the root DFHack
+folder, and will be installed to ``hack/docs`` when you next install DFHack in a
+DF folder.
+
+FAQ
+---
+
+**What do I do when I'm getting the error** ``fatal error C1083: Cannot open include file: 'CoreProtocol.pb.h': No such file or directory``
+**or** ``..\zlib\lib\zlib.lib : warning LNK4272: library machine type 'X86' conflicts with target machine type 'x64'``?
+
+* Check the architecture CMake has configured: ``cat CMakeCache.txt | grep ARCH`` if it is 32, and you're building 64 [or vice versa] you'll need to regenerate the project from scratch.
+
+* Modify the cmake command to include ``-DDFHACK_BUILD_ARCH=64`` or 32 if that's the appropriate value for what you're building. If you didn't use cmake, check the batch file for the cmake command.
+
+
+**What should I do if running one of the generate-MSVC- batch files I get an error** ``Failed to run MSBuild command:``?
+
+Modify the batch file, it may be that you haven't installed Visual Studio or that you haven't installed the version used by default as the generator inside the batch files. Simply remove the -G generator option from the cmake command and try again. Remember to delete VC2015/ before running the modified batch file.
 
-Building/installing from the Visual Studio IDE:
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-After running the CMake generate script you will have a new folder called VC2015
-or VC2015_32, depending on the architecture you specified. Open the file
-``dfhack.sln`` inside that folder. If you have multiple versions of Visual
-Studio installed, make sure you open with Visual Studio 2015.
 
-The first thing you must then do is change the build type. It defaults to Debug,
-but this cannot be used on Windows. Debug is not binary-compatible with DF.
-If you try to use a debug build with DF, you'll only get crashes and for this
-reason the Windows "debug" scripts actually do RelWithDebInfo builds.
-After loading the Solution, change the Build Type to either ``Release``
-or ``RelWithDebInfo``.
+**What does it mean if building or running cmake I see the error** ``Can't locate XML/LibXML.pm in @INC (you may need to
+install the XML::LibXML module)``?
 
-Then build the ``INSTALL`` target listed under ``CMakePredefinedTargets``.
+* It means perl can't find that dependency.
+* You'll first want to check if cmake finds perl at all
 
+  * then that the path of perl matches what you installed, or if it is perhaps the git-bash packaged perl
+  * cmake output ex: ``-- Found Perl: C:/cmder/vendor/git-for-windows/usr/bin/perl.exe (found version "5.34.0")``
+* If it isn't the version you expect to see, read the section about `git bash`.
+* If it is, ensure the LibXML and LibXLST dependencies are installed for the detected perl installation.
+* If they are, seek help on `discord`_.
 
-Building the documentation
-==========================
+.. _discord: https://dfhack.org/discord
```

Comment:
I would suggest linking to the support doc with

```
`seek help <support>`
```

or similar. That has links to all of our support channels and is a single place that gets updated.

Reply:
Too bad you didn't make this as a code suggestion I could accept right now.

Reply:
oh, lame